### PR TITLE
Adding a CLI entrypoint to the V4 validation logic

### PR DESCRIPTION
**Pull request by scottynomad**
See [CHANGELOG](project/parcel/CHANGELOG.md) for details.  Drove some complexity out of the schema to get more concise error reports.

### CODE REVIEW

File: packages/project/scripts/validate.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+function validateV4(
+  projectJson: Project
+): ReturnType<typeof validateProjectData> {
+  const validatonResult = validateProjectData(projectJson);
+  return validatonResult;
+}
+
+export const main = (): void => {
+  const input = fs.readFileSync(0, 'utf-8');
```
Comment (mrdave-dev) [inline packages/project/scripts/validate.ts:15]: Whoa. This works? Everything I've seen says you should read from `process.stdin.fd`

Comment (scottynomad) [reply packages/project/scripts/validate.ts:15]: https://nodejs.org/api/process.html#processstdinfd

Comment (scottynomad) [reply packages/project/scripts/validate.ts:15]: It's certainly more reader friendly to use `process.stdin.fd`.  It was just causing typescript problems I was avoiding.

Comment (mrdave-dev) [reply packages/project/scripts/validate.ts:15]: Ah cool

File: packages/project/scripts/validateIsolated.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const ajv = addFormats(
+    new Ajv({
+      useDefaults: 'empty',
+      allErrors: true,
+      inlineRefs: false,
+      $data: true,
+      allowUnionTypes: true,
+    })
+  );
+  // Ignore the custom NORI errorMessage and validationRules clauses
```
Comment (jaycenhorton) [inline packages/project/scripts/validateIsolated.ts:33]: might be nice to just `console.warn("Note that this validation library does not currently validate the `validationRule` or `errorMessage` fields");` for external consumers

**Review by jaycenhorton — APPROVED** [review]

**Pull request merged**